azure_functions_worker/dispatcher.py

```diff
@@ -526,12 +535,22 @@ async def _handle__close_shared_memory_resources_request(self, req):
         results = {mem_map_name: False for mem_map_name in map_names}
 
         try:
-            for mem_map_name in map_names:
+            for map_name in map_names:
                 try:
-                    success = self._shmem_mgr.free_mem_map(mem_map_name)
-                    results[mem_map_name] = success
+                    if self._function_data_cache_enabled:
+                        # If the cache is enabled, let the host decide when to
+                        # delete the resources.
+                        # Just drop the reference from the worker.
+                        to_delete = False
+                    else:
+                        # If the cache is not enabled, the worker should free
+                        # the resources as at this point the host has read the
+                        # memory maps and does not need them.
+                        to_delete = True
```

Comment:
```suggestion
                    to_delete_reference = False if self._function_data_cache_enabled else True
```

And please add the comment in the python docs section of this method itself.